great_expectations/experimental/datasources/interfaces.py

```diff
@@ -65,21 +75,103 @@ def datasource(self) -> Datasource:
         return self._datasource
 
     # TODO (kilo): remove setter and add custom init for DataAsset to inject datasource in constructor??
+    # This setter is non-functional: https://github.com/pydantic/pydantic/issues/3395
+    # There is some related discussion linked from that ticket which may be a workaround.
     @datasource.setter
     def datasource(self, ds: Datasource):
         assert isinstance(ds, Datasource)
         self._datasource = ds
 
-    def get_batch_request(self, options: Optional[BatchRequestOptions]) -> BatchRequest:
+    # Abstract Methods
+    def batch_request_options_template(
+        self,
+    ) -> BatchRequestOptions:
+        """A BatchRequestOptions template for get_batch_request.
+
+        Returns:
+            A BatchRequestOptions dictionary with the correct shape that get_batch_request
+            will understand. All the option values are defaulted to None.
+        """
+        raise NotImplementedError
+
+    def get_batch_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> List[Batch]:
         raise NotImplementedError
 
+    # End Abstract Methods
 
+    def get_batch_request(
```

Comment:
The methods here use to live in `TableAsset` but were generic enough that I pulled them into `DataAsset`.